.ci/pitest-suppressions/pitest-common-suppressions.xml

```diff
@@ -17,4 +377,3712 @@
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to java/util/Map::put</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;EmptyBlockCheck&quot;,</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>PackageObjectFactory.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.ArgumentPropagationMutator</mutator>
+    <description>replaced call to java/util/Map::put with argument</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;EmptyBlockCheck&quot;,</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>PackageObjectFactory.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to java/util/Map::put</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;EmptyCatchBlockCheck&quot;,</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>PackageObjectFactory.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.ArgumentPropagationMutator</mutator>
+    <description>replaced call to java/util/Map::put with argument</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;EmptyCatchBlockCheck&quot;,</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>PackageObjectFactory.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to java/util/Map::put</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;LeftCurlyCheck&quot;,</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>PackageObjectFactory.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
+    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.ArgumentPropagationMutator</mutator>
+    <description>replaced call to java/util/Map::put with argument</description>
+    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;LeftCurlyCheck&quot;,</lineContent>
+  </mutation>
```

Comment:
@nick-mancuso This mutation is surviving, but it is getting killed in local-
```
└─$ git diff origin/master -- src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java                                                                             1 ⚙
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java b/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
index 568abdb44..17f863263 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
@@ -445,8 +445,9 @@ public class PackageObjectFactory implements ModuleFactory {
      * Fill short-to-full module names map with Checks from blocks package.
      */
     private static void fillChecksFromBlocksPackage() {
-        NAME_TO_FULL_MODULE_NAME.put("AvoidNestedBlocksCheck",
-                BASE_PACKAGE + ".checks.blocks.AvoidNestedBlocksCheck");
+        String var10002 = "AvoidNestedBlocksCheck";
+        Map var10001 = NAME_TO_FULL_MODULE_NAME;
+        String var10000 = "com.puppycrawl.tools.checkstyle.checks.blocks.AvoidNestedBlocksCheck";
         NAME_TO_FULL_MODULE_NAME.put("EmptyBlockCheck",
                 BASE_PACKAGE + ".checks.blocks.EmptyBlockCheck");
         NAME_TO_FULL_MODULE_NAME.put("EmptyCatchBlockCheck",
```
```
└─$ mvn test -Dtest=PackageObjectFactoryTest test
[INFO] Running com.puppycrawl.tools.checkstyle.PackageObjectFactoryTest
[ERROR] Tests run: 25, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 45.644 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.PackageObjectFactoryTest
[ERROR] testNameToFullModuleNameMap  Time elapsed: 36.891 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts: 
Invalid canonical name: Optional[class com.puppycrawl.tools.checkstyle.checks.blocks.AvoidNestedBlocksCheck]
value of: isPresent()
expected to be false
        at com.puppycrawl.tools.checkstyle.PackageObjectFactoryTest.testNameToFullModuleNameMap(PackageObjectFactoryTest.java:391)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   PackageObjectFactoryTest.testNameToFullModuleNameMap:391 Invalid canonical name: Optional[class com.puppycrawl.tools.checkstyle.checks.blocks.AvoidNestedBlocksCheck]
value of: isPresent()
expected to be false
[INFO] 
[ERROR] Tests run: 25, Failures: 1, Errors: 0, Skipped: 0
```

Reply:
The test fails in CI too (with hardcoded mutation), see https://github.com/checkstyle/checkstyle/pull/11996/checks?check_run_id=7541868988

Reply:
This could one of a two different issues (or a combination :) ):

1. This is not the correct hardcoded mutation (does not correspond to the bytecode)
2. there is some bug in pitest (see https://github.com/hcoles/pitest/issues/677)

Let's figure out what's going on here. @rnveach any thoughts?

Reply:
> This is not the correct hardcoded mutation (does not correspond to the bytecode)

No, this is the correct hardcoded mutation, I don't hardcode the mutation myself, I have modified the pitest repo a bit for it to generate mutated bytecode for me, and I decompile the mutated bytecode using the fernflower decompiler. It is very unlikely fernflower is giving me the wrong result.

Reply:
I doubt it is due to 2, if the hardcoded mutation is correct, and the test is failing in local as well as CI, also if we forget about this mutation for a while, and see the mutation-
```XML
  <mutation unstable="false">
    <sourceFile>PackageObjectFactory.java</sourceFile>
    <mutatedClass>com.puppycrawl.tools.checkstyle.PackageObjectFactory</mutatedClass>
    <mutatedMethod>fillChecksFromBlocksPackage</mutatedMethod>
    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
    <description>removed call to java/util/Map::put</description>
    <lineContent>NAME_TO_FULL_MODULE_NAME.put(&quot;LeftCurlyCheck&quot;,</lineContent>
  </mutation>
```
Removed call to `put(..)` is equivalent to not putting that value in that map, or just simply commenting out the line where `put(..)` is used. If we comment it in the source code, then also 
`PackageObjectFactoryTest#testNameToFullModuleNameMap()` fails.

One thing here is attracting my attention, `PackageObjectFactoryTest#testNameToFullModuleNameMap()` takes anywhere between `444 ms to 14 sec` to run in my local (based on whether it has to be compiled again or uses cached value), I believe `timeoutFactor` and `timeoutConstant` might be playing a role here which results in this mutation being considered as survived.

Reply:
I don't think there is anything wrong with pitest, I think it is our tests that result in these mutations surviving. I did a run with all other tests commented out in `PackageObjectFactoryTest` except
`PackageObjectFactoryTest#testNameToFullModuleNameMap()` and the mutations under discussion now were killed.

See https://vyom-yadav.github.io/pitest-all-latest/pitest-common-modified/index.html

Reply:
Interesting results, the mutation gets killed only when no other test accesses `PackageObjectFactory`, not sure why this is happening, maybe due to some mutator, @rnveach do you have any idea regarding this?

Reply:
> I decompile the mutated bytecode using the fernflower decompiler. 

Please share entire class here.

Reply:
@Vyom-Yadav let's merge this and investigate further. I added comment to issue, and we will be blocked by this for every new check that we add, so we will not forget.